ingestion/src/metadata/ingestion/source/dashboard/mode.py

```diff
@@ -0,0 +1,231 @@
+                        metadata=self.metadata,
+                        database=data_source.get("database"),
+                        service_name=self.source_config.dbServiceName,
+                        database_schema=database_schema_name,
+                        table=table,
+                    )
+                    for from_entity in from_entities:
+                        to_entity = self.metadata.get_by_name(
+                            entity=Lineage_Dashboard,
+                            fqn=fqn.build(
+                                self.metadata,
+                                Lineage_Dashboard,
+                                service_name=self.config.serviceName,
+                                dashboard_name=dashboard_details.get("token"),
+                            ),
+                        )
+                        if from_entity and to_entity:
+                            lineage = AddLineageRequest(
+                                edge=EntitiesEdge(
+                                    fromEntity=EntityReference(
+                                        id=from_entity.id.__root__, type="table"
+                                    ),
+                                    toEntity=EntityReference(
+                                        id=to_entity.id.__root__, type="dashboard"
+                                    ),
+                                )
+                            )
+                            yield lineage
+        except Exception as err:  # pylint: disable=broad-except
+            logger.debug(traceback.format_exc())
+            logger.error(err)
+
+    def fetch_dashboard_charts(
+        self, dashboard_details: dict
+    ) -> Iterable[CreateChartRequest]:
+        """Get chart method
+        Args:
+            dashboard_details:
+        Returns:
+            Iterable[Chart]
+        """
+        self.charts = []
+        response_queries = self.client.get_all_queries(
+            workspace_name=self.workspace_name,
+            report_token=dashboard_details.get("token"),
+        )
+        queries = response_queries["_embedded"]["queries"]
+        for query in queries:
+            response_charts = self.client.get_all_charts(
+                workspace_name=self.workspace_name,
+                report_token=dashboard_details.get("token"),
+                query_token=query.get("token"),
+            )
+            charts = response_charts["_embedded"]["charts"]
+            for chart in charts:
+                try:
+                    if filter_by_chart(
+                        self.source_config.chartFilterPattern,
+                        chart["view_vegas"]["title"],
+                    ):
+                        self.status.filter(
+                            chart["view_vegas"]["title"], "Chart Pattern not Allowed"
+                        )
+                        continue
+                    chart_url = (
+                        f"{self.base_url}"
```

Comment:
what if we define the `self.base_url` as `hostPort` with the default value, that way we might not break UI 🤔 @pmbrull

Reply:
If it is something not really configurable that might be a bit confusing, but I see we have done that for PowerBI, so yeah thinking about it again it can be a possibility